src/main/java/seedu/address/logic/commands/BuyCommand.java

```diff
@@ -50,6 +61,30 @@ public BuyCommand(Good boughtGood) {
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
+        // supplier could possibly not exist
+        Name supplierNameToFind = boughtGood.getSupplierName();
+        int numMatchingSuppliers = (int) model.getFilteredSupplierList().stream()
+                .filter(supplier -> supplier.getName().equals(supplierNameToFind)).count();
+        if (numMatchingSuppliers <= 0) {
+            throw new CommandException(MESSAGE_BUYING_FROM_NONEXISTENT_SUPPLIER);
```

Comment:
Would it be easier to buy [index of supplier] rather than typing the exact name of the supplier?

Reply:
I was considering that as well. The implications is that we need to add a prefix for index.

The add-s [index] or delete-g [index] work well because they only accept 1 argument.

I think we can discuss this during the coming tutorial.

Reply:
there is no need for index prefix.
all command follows this structure: [keyword][space][preamble][space][prefix/xxx][space][prefix/xxx]
you can easily obtain index under preamble, you may want to check out editCommandParser

Reply:
You are right, i missed that piece of functionality. Thanks for pointing out, i think that can be used to solve this problem.